### ISSUE TITLE
NSInvalidArgumentException · *** -[__NSArrayM insertObject:atIndex:]: object cannot be nil

### ISSUE THREAD

**Issue by SudoPlz** [Instabug/Instabug-iOS#170]
Stack trace:

```Objective-C
NSInvalidArgumentException *** -[__NSArrayM insertObject:atIndex:]: object cannot be nil 
    Frameworks/CoreFoundation.framework/CoreFoundation ___exceptionPreprocess
    /usr/lib/libobjc.A.dylib objc_exception_throw
    Frameworks/CoreFoundation.framework/CoreFoundation -[__NSArrayM insertObject:atIndex:]
    Instabug/Logging/IBGDataStore.m:576:9 -[IBGDataStore userEventLogs]
    Instabug/Caching/IBGSession.m:80:80 -[IBGSession requestParameters]
    Instabug/Caching/IBGSession.m:131:87 -[IBGSession send]
    Instabug/Caching/IBGSession.m:67:5 -[IBGSession persistAndSend]
    Instabug/Managers/IBGSessionManager.m:74:5 -[IBGSessionManager sendSessionRequest]
    Frameworks/CoreFoundation.framework/CoreFoundation ___CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__
    Frameworks/CoreFoundation.framework/CoreFoundation __CFXRegistrationPost
    Frameworks/CoreFoundation.framework/CoreFoundation ____CFXNotificationPost_block_invoke
    Frameworks/CoreFoundation.framework/CoreFoundation -[_CFXNotificationRegistrar find:object:observer:enumerator:]
    Frameworks/CoreFoundation.framework/CoreFoundation __CFXNotificationPost
    Frameworks/Foundation.framework/Foundation -[NSNotificationCenter postNotificationName:object:userInfo:]
    Frameworks/UIKit.framework/UIKit ___47-[UIApplication _applicationDidEnterBackground]_block_invoke
    Frameworks/UIKit.framework/UIKit +[UIViewController _performWithoutDeferringTransitions:]
    Frameworks/UIKit.framework/UIKit -[UIApplication _applicationDidEnterBackground]
    Frameworks/UIKit.framework/UIKit -[UIApplication _handleApplicationDeactivationWithScene:shouldForceExit:transitionContext:completion:]
    Frameworks/UIKit.framework/UIKit -[UIApplication _handleApplicationLifecycleEventWithScene:transitionContext:completion:]
    Frameworks/UIKit.framework/UIKit ___70-[UIApplication scene:didUpdateWithDiff:transitionContext:completion:]_block_invoke
    Frameworks/UIKit.framework/UIKit -[UIApplication scene:didUpdateWithDiff:transitionContext:completion:]
    Frameworks/UIKit.framework/UIKit -[UIApplicationSceneClientAgent scene:handleEvent:withCompletion:]
    PrivateFrameworks/FrontBoardServices.framework/FrontBoardServices ___80-[FBSSceneImpl updater:didUpdateSettings:withDiff:transitionContext:completion:]_block_invoke.376
    PrivateFrameworks/FrontBoardServices.framework/FrontBoardServices ___FBSSERIALQUEUE_IS_CALLING_OUT_TO_A_BLOCK__
    PrivateFrameworks/FrontBoardServices.framework/FrontBoardServices -[FBSSerialQueue _performNext]
    PrivateFrameworks/FrontBoardServices.framework/FrontBoardServices -[FBSSerialQueue _performNextFromRunLoopSource]
    Frameworks/CoreFoundation.framework/CoreFoundation ___CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__
    Frameworks/CoreFoundation.framework/CoreFoundation ___CFRunLoopDoSources0
    Frameworks/CoreFoundation.framework/CoreFoundation ___CFRunLoopRun
    Frameworks/CoreFoundation.framework/CoreFoundation CFRunLoopRunSpecific
    PrivateFrameworks/GraphicsServices.framework/GraphicsServices GSEventRunModal
    Frameworks/UIKit.framework/UIKit UIApplicationMain
    TestApp mh_execute_header
    /usr/lib/system/libdyld.dylib start
```

And another one (same error):

```Objective-C
NSInvalidArgumentException *** -[__NSArrayM insertObject:atIndex:]: object cannot be nil 
    Frameworks/CoreFoundation.framework/CoreFoundation ___exceptionPreprocess
    /usr/lib/libobjc.A.dylib objc_exception_throw
    Frameworks/CoreFoundation.framework/CoreFoundation -[__NSArrayM insertObject:atIndex:]
    Instabug/Logging/IBGDataStore.m:576:9 -[IBGDataStore userEventLogs]
    Instabug/Caching/IBGSession.m:80:80 -[IBGSession requestParameters]
    Instabug/Caching/IBGSession.m:131:87 -[IBGSession send]
    Instabug/Caching/IBGSession.m:67:5 -[IBGSession persistAndSend]
    Instabug/Managers/IBGSessionManager.m:74:5 -[IBGSessionManager sendSessionRequest]
    Frameworks/CoreFoundation.framework/CoreFoundation ___CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__
    Frameworks/CoreFoundation.framework/CoreFoundation __CFXRegistrationPost
    Frameworks/CoreFoundation.framework/CoreFoundation ____CFXNotificationPost_block_invoke
    Frameworks/CoreFoundation.framework/CoreFoundation -[_CFXNotificationRegistrar find:object:observer:enumerator:]
    Frameworks/CoreFoundation.framework/CoreFoundation __CFXNotificationPost
    Frameworks/Foundation.framework/Foundation -[NSNotificationCenter postNotificationName:object:userInfo:]
    Frameworks/UIKit.framework/UIKit ___47-[UIApplication _applicationDidEnterBackground]_block_invoke
    Frameworks/UIKit.framework/UIKit +[UIViewController _performWithoutDeferringTransitions:]
    Frameworks/UIKit.framework/UIKit -[UIApplication _applicationDidEnterBackground]
    Frameworks/UIKit.framework/UIKit -[UIApplication _handleApplicationDeactivationWithScene:shouldForceExit:transitionContext:completion:]
    Frameworks/UIKit.framework/UIKit -[UIApplication _handleApplicationLifecycleEventWithScene:transitionContext:completion:]
    Frameworks/UIKit.framework/UIKit ___70-[UIApplication scene:didUpdateWithDiff:transitionContext:completion:]_block_invoke
    Frameworks/UIKit.framework/UIKit -[UIApplication scene:didUpdateWithDiff:transitionContext:completion:]
    Frameworks/UIKit.framework/UIKit -[UIApplicationSceneClientAgent scene:handleEvent:withCompletion:]
    PrivateFrameworks/FrontBoardServices.framework/FrontBoardServices ___80-[FBSSceneImpl updater:didUpdateSettings:withDiff:transitionContext:completion:]_block_invoke.376
    PrivateFrameworks/FrontBoardServices.framework/FrontBoardServices ___FBSSERIALQUEUE_IS_CALLING_OUT_TO_A_BLOCK__
    PrivateFrameworks/FrontBoardServices.framework/FrontBoardServices -[FBSSerialQueue _performNext]
    PrivateFrameworks/FrontBoardServices.framework/FrontBoardServices -[FBSSerialQueue _performNextFromRunLoopSource]
    Frameworks/CoreFoundation.framework/CoreFoundation ___CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__
    Frameworks/CoreFoundation.framework/CoreFoundation ___CFRunLoopDoSources0
    Frameworks/CoreFoundation.framework/CoreFoundation ___CFRunLoopRun
    Frameworks/CoreFoundation.framework/CoreFoundation CFRunLoopRunSpecific
    PrivateFrameworks/GraphicsServices.framework/GraphicsServices GSEventRunModal
    Frameworks/UIKit.framework/UIKit UIApplicationMain
    TestApp mh_execute_header
    /usr/lib/system/libdyld.dylib start
```

Threads:

```
Thread 1·
/usr/lib/system/libsystem_kernel.dylibmach_msg_trap	
/usr/lib/system/libsystem_kernel.dylibmach_msg	
Frameworks/CoreFoundation.framework/CoreFoundation___CFRunLoopServiceMachPort	
Frameworks/CoreFoundation.framework/CoreFoundation___CFRunLoopRun	
Frameworks/CoreFoundation.framework/CoreFoundationCFRunLoopRunSpecific	
Frameworks/Foundation.framework/Foundation-[NSRunLoop(NSRunLoop) runMode:beforeDate:]	
Frameworks/Foundation.framework/Foundation-[NSRunLoop(NSRunLoop) runUntilDate:]	
Frameworks/UIKit.framework/UIKit-[UIEventFetcher threadMain]	
Frameworks/Foundation.framework/Foundation___NSThread__start__	
/usr/lib/system/libsystem_pthread.dylib__pthread_body	
/usr/lib/system/libsystem_pthread.dylib__pthread_start	
Thread 2·
/usr/lib/system/libsystem_kernel.dylibmach_msg_trap	
/usr/lib/system/libsystem_kernel.dylibmach_msg	
Frameworks/CoreFoundation.framework/CoreFoundation___CFRunLoopServiceMachPort	
Frameworks/CoreFoundation.framework/CoreFoundation___CFRunLoopRun	
Frameworks/CoreFoundation.framework/CoreFoundationCFRunLoopRunSpecific	
Frameworks/CFNetwork.framework/CFNetwork+[NSURLConnection(Loader) _resourceLoadLoop:]	
Frameworks/Foundation.framework/Foundation___NSThread__start__	
/usr/lib/system/libsystem_pthread.dylib__pthread_body	
/usr/lib/system/libsystem_pthread.dylib__pthread_start	
Thread 3·
/usr/lib/system/libsystem_kernel.dylibmach_msg_trap	
/usr/lib/system/libsystem_kernel.dylibmach_msg	
Frameworks/CoreFoundation.framework/CoreFoundation___CFRunLoopServiceMachPort	
Frameworks/CoreFoundation.framework/CoreFoundation___CFRunLoopRun	
Frameworks/CoreFoundation.framework/CoreFoundationCFRunLoopRunSpecific	
AcuityApp__cxa_throw	
 Frameworks/Foundation.framework/Foundation___NSThread__start__	
/usr/lib/system/libsystem_pthread.dylib__pthread_body	
/usr/lib/system/libsystem_pthread.dylib__pthread_start	
Thread 4·
/usr/lib/system/libsystem_kernel.dylib___semwait_signal	
/usr/lib/system/libsystem_c.dylibnanosleep	
/usr/lib/libc++.1.dylib__ZNSt3__111this_thread9sleep_forERKNS_6chrono8durationIxNS_5ratioILl1ELl1000000000EEEEE	
Frameworks/JavaScriptCore.framework/JavaScriptCore__ZNSt3__111this_thread9sleep_forIxNS_5ratioILl1ELl1000EEEEEvRKNS_6chrono8durationIT_T0_EE	
Frameworks/JavaScriptCore.framework/JavaScriptCore__ZN7bmallocL14waitUntilFalseERNSt3__111unique_lockINS_11StaticMutexEEENS0_6chrono8durationIxNS0_5ratioILl1ELl1000EEEEERb	
Frameworks/JavaScriptCore.framework/JavaScriptCore__ZN7bmalloc4Heap8scavengeERNSt3__111unique_lockINS_11StaticMutexEEENS1_6chrono8durationIxNS1_5ratioILl1ELl1000EEEEE	
Frameworks/JavaScriptCore.framework/JavaScriptCore__ZN7bmalloc4Heap18concurrentScavengeEv	
Frameworks/JavaScriptCore.framework/JavaScriptCore__ZN7bmalloc9AsyncTaskINS_4HeapEMS1_FvvEE13threadRunLoopEv	
Frameworks/JavaScriptCore.framework/JavaScriptCore__ZN7bmalloc9AsyncTaskINS_4HeapEMS1_FvvEE16threadEntryPointEPS4_	
Frameworks/JavaScriptCore.framework/JavaScriptCore__ZNSt3__114__thread_proxyINS_5tupleIJPFvPN7bmalloc9AsyncTaskINS2_4HeapEMS4_FvvEEEES8_EEEEEPvSC_	
/usr/lib/system/libsystem_pthread.dylib__pthread_body	
/usr/lib/system/libsystem_pthread.dylib__pthread_start	
Thread 5·
/usr/lib/system/libsystem_kernel.dylib___workq_kernreturn	
/usr/lib/system/libsystem_pthread.dylib__pthread_wqthread	
Thread 6·
/usr/lib/system/libsystem_kernel.dylib___workq_kernreturn	
/usr/lib/system/libsystem_pthread.dylib__pthread_wqthread	
Thread 7·
/usr/lib/system/libsystem_kernel.dylibmach_msg_trap	
/usr/lib/system/libsystem_kernel.dylibmach_msg	
/usr/lib/system/libsystem_kernel.dylibthread_suspend	
AcuityApp__cxa_throw	
 /usr/lib/system/libsystem_pthread.dylib__pthread_body	
/usr/lib/system/libsystem_pthread.dylib__pthread_start	
Thread 8·
/usr/lib/system/libsystem_kernel.dylibmach_msg_trap	
/usr/lib/system/libsystem_kernel.dylibmach_msg	
AcuityApp__cxa_throw	
 /usr/lib/system/libsystem_pthread.dylib__pthread_body	
/usr/lib/system/libsystem_pthread.dylib__pthread_start	
Thread 9·
/usr/lib/system/libsystem_kernel.dylibmach_msg_trap	
/usr/lib/system/libsystem_kernel.dylibmach_msg	
Frameworks/CoreFoundation.framework/CoreFoundation___CFRunLoopServiceMachPort	
Frameworks/CoreFoundation.framework/CoreFoundation___CFRunLoopRun	
Frameworks/CoreFoundation.framework/CoreFoundationCFRunLoopRunSpecific	
Frameworks/CFNetwork.framework/CFNetwork__ZL27_privateRunloopEmulationSetPv	
/usr/lib/system/libsystem_pthread.dylib__pthread_body	
/usr/lib/system/libsystem_pthread.dylib__pthread_start	
Thread 10·
Frameworks/CoreFoundation.framework/CoreFoundation___CFSearchStringROM	
Frameworks/CoreFoundation.framework/CoreFoundation___CFSearchStringROM	
Frameworks/CoreFoundation.framework/CoreFoundation___CFStringCreateImmutableFunnel3	
Frameworks/CoreFoundation.framework/CoreFoundationCFStringCreateWithBytes	
Frameworks/Foundation.framework/FoundationnewJSONString	
Frameworks/Foundation.framework/FoundationnewJSONValue	
Frameworks/Foundation.framework/FoundationnewJSONValue	
Frameworks/Foundation.framework/FoundationnewJSONValue	
Frameworks/Foundation.framework/Foundation-[_NSJSONReader parseUTF8JSONData:skipBytes:options:]	
Frameworks/Foundation.framework/Foundation-[_NSJSONReader parseData:options:]	
Frameworks/Foundation.framework/Foundation+[NSJSONSerialization JSONObjectWithData:options:error:]	
AcuityAppmh_execute_header	
 AcuityAppmh_execute_header	
 /usr/lib/system/libdispatch.dylib__dispatch_client_callout	
/usr/lib/system/libdispatch.dylib__dispatch_barrier_sync_f_invoke	
AcuityAppmh_execute_header	
 AcuityAppmh_execute_header	
 AcuityAppmh_execute_header	
 AcuityAppmh_execute_header	
 AcuityAppmh_execute_header	
 Frameworks/Foundation.framework/Foundation___NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__	
Frameworks/Foundation.framework/Foundation-[NSBlockOperation main]	
Frameworks/Foundation.framework/Foundation-[__NSOperationInternal _start:]	
Frameworks/Foundation.framework/Foundation___NSOQSchedule_f	
/usr/lib/system/libdispatch.dylib__dispatch_client_callout	
/usr/lib/system/libdispatch.dylib__dispatch_queue_serial_drain	
/usr/lib/system/libdispatch.dylib__dispatch_queue_invoke	
/usr/lib/system/libdispatch.dylib__dispatch_root_queue_drain	
/usr/lib/system/libdispatch.dylib__dispatch_worker_thread3	
/usr/lib/system/libsystem_pthread.dylib__pthread_wqthread	
Thread 11·
/usr/lib/system/libsystem_kernel.dylib___workq_kernreturn	
/usr/lib/system/libsystem_pthread.dylib__pthread_wqthread	
Thread 12·
/usr/lib/system/libsystem_kernel.dylib___workq_kernreturn	
/usr/lib/system/libsystem_pthread.dylib__pthread_wqthread	
Thread 13·
/usr/lib/system/libsystem_kernel.dylib___workq_kernreturn	
/usr/lib/system/libsystem_pthread.dylib__pthread_wqthread
```

I'm using `react-native-instabug` and I'm installing Instabug via cocoapods like so:

`pod 'Instabug', '~> 7.3.2'`

**Comment by Kmohamed** [comment]
Let me check it and get back to you soon

**Comment by Kmohamed** [comment]
@SudoPlz we are very sorry for inconvenience, it will be fixed next release.

**Comment by SudoPlz** [comment]
No problem, let me know once it's fixed. Thank you @Kmohamed

**Comment by Kmohamed** [comment]
sure 👍

**Comment by Kmohamed** [comment]
Released on version 7.3.6 🎉
